### ISSUE TITLE
Allow availability overrides based on event name patterns and presence of attendees

### ISSUE THREAD

**Issue by pjatx** [calcom/cal.com#14614]
### Is your proposal related to a problem?

I proactively create all sorts of recurring time blocks that could change on a daily or weekly basis to prevent internal meetings or disruptions to derail my day.

Think lunch, deep work, morning communications, afternoon communications, etc. [Savvy Cal](https://savvycal.com) has a super useful feature that allows you to create "Time Blocks" on the availability level which are basically just search patterns. If matched, they will show you as available even though your calendar might be blocked with an event marked busy. 

This is super useful for example if you want to block of time each day for lunch or answering emails, but you would also like external guests to be able to schedule a lunch, happy hours, whatever with you.

This is the only reason I'm not moving all of my scheduling fully over to Cal.com.

### Describe the solution you'd like

Open to different solutions, but I think the event name / pattern matching works super well in my case. 

### Describe alternatives you've considered

I've tried to use the date overrides, but the value in this feature is for it to be flexible based on whenever I decide to block my calendar using a specific name. 

### Additional context

(Write your answer here.)

### Requirement/Document

![image](https://github.com/calcom/cal.com/assets/5004634/62ecb087-9b7e-4a72-bee1-6521fd924c05)

![image](https://github.com/calcom/cal.com/assets/5004634/530208b6-f8ac-4a93-a723-9b0b4b139205)

---
##### House rules
- If this issue has a `🚨 needs approval` label, don't start coding yet. Wait until a core member approves feature request by removing this label, then you can start coding.
  - For clarity: Non-core member issues automatically get the `🚨 needs approval` label.
  - Your feature ideas are invaluable to us! However, they undergo review to ensure alignment with the product's direction.

**Comment by ciaranha** [comment]
Duplicate of https://github.com/calcom/cal.com/issues/13305
